### PR TITLE
osd: mds: suggest clock skew when failing to obtain rotating service keys

### PR DESCRIPTION
**Pull request by gregsfortytwo**
This short PR just suggests a course of action to users, as it comes up fairly often on the mailing list.

Fixes: https://tracker.ceph.com/issues/17170

Signed-off-by: Greg Farnum <gfarnum@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by ajarr — APPROVED** [review]

File: src/mds/MDSDaemon.cc [hunk 1]
```diff
@@ -552,7 +552,8 @@ int MDSDaemon::init()
       continue;
     }
     derr << "ERROR: failed to refresh rotating keys, "
-         << "maximum retry time reached." << dendl;
+         << "maximum retry time reached."
+	 << " Maybe I have a clock skew against the monitors?" << dendl;
```
Comment (rzarzynski) [inline src/mds/MDSDaemon.cc:556]: nit: mixed spaces and tabs?

File: src/osd/OSD.cc [hunk 2]
```diff
@@ -3824,7 +3824,8 @@ int OSD::init()
     derr << "unable to obtain rotating service keys; retrying" << dendl;
     ++rotating_auth_attempts;
     if (rotating_auth_attempts > g_conf()->max_rotating_auth_attempts) {
-        derr << __func__ << " wait_auth_rotating timed out" << dendl;
+        derr << __func__ << " wait_auth_rotating timed out"
+	     <<" -- maybe I have a clock skew against the monitors?" << dendl;
```
Comment (rzarzynski) [inline src/osd/OSD.cc:3828]: Same here.

File: src/mds/MDSDaemon.cc (hunk 1 above)
Comment (gregsfortytwo) [reply src/mds/MDSDaemon.cc:556]: I'm using emacs and the formatting there is controlled by the first line in the file, so if we want that to stop happening we'd need to agree on and commit updates to those.

**Review by vshankar — APPROVED** [review]

**Review by dparmar18 — APPROVED** [review]

**Comment by vshankar** [comment]
@gregsfortytwo I see you added needs-qa tag. Seems to me that this is good to merge - do we want to run it through integration tests?

**Comment by gregsfortytwo** [comment]
> @gregsfortytwo I see you added needs-qa tag. Seems to me that this is good to merge - do we want to run it through integration tests?

There's no particular hurry and we should always test things before they merge? For all I know there's some test that will complain because the strings are wrong now!

**Comment by vshankar** [comment]
> > @gregsfortytwo I see you added needs-qa tag. Seems to me that this is good to merge - do we want to run it through integration tests?
> 
> There's no particular hurry and we should always test things before they merge? For all I know there's some test that will complain because the strings are wrong now!

In general, yes. Although I merge such changes (since they are unlikely to break tests, but we can always adjust the tests when we run into failures later).

**Comment by ljflores** [comment]
Rados suite review: http://pulpito.front.sepia.ceph.com/?branch=wip-yuri7-testing-2022-10-17-0814

Failures:, unrelated
    1. https://tracker.ceph.com/issues/57311
    2. https://tracker.ceph.com/issues/52321
    3. https://tracker.ceph.com/issues/52657
    4. https://tracker.ceph.com/issues/57935

Details:
    1. rook: ensure CRDs are installed first - Ceph - Orchestrator
    2. qa/tasks/rook times out: 'check osd count' reached maximum tries (90) after waiting for 900 seconds - Ceph - Orchestrator
    2. rados/thrash-erasure-code: wait_for_recovery timeout due to "active+clean+remapped+laggy" pgs - Ceph - RADOS
    3. MOSDPGLog::encode_payload(uint64_t): Assertion `HAVE_FEATURE(features, SERVER_NAUTILUS)' - Ceph - RADOS
    4. all test jobs get stuck at "Running task ansible.cephlab..." - Infrstructure - Sepia

**Comment by vshankar** [comment]
@yuriw What happened to the fs suite jobs for this change (after all the teuthology mess in November)?

**Comment by ljflores** [comment]
@vshankar the fs jobs need reviewing here: https://trello.com/c/hdiNA6Zq/1651-wip-yuri7-testing-2022-11-18-1548-old-wip-yuri7-testing-2022-10-17-0814

**Review by badone — APPROVED** [review]

**Review by vshankar — APPROVED** [review]

**Pull request merged**